apps/src/sites/studio/pages/components/editorRedux.js

```diff
@@ -0,0 +1,127 @@
+/* global levelKeyList */
+
+import _ from 'lodash';
+
+function updatePositions(node) {
+  for (var i = 0; i < node.length; i++) {
+    node[i].position = i + 1;
+  }
+}
+
+let newStageId = -1;
+let newLevelId = -1;
+
+export default function reducer(state, action) {
+  let newState = _.cloneDeep(state);
+
+  switch (action.type) {
+    case 'REORDER_LEVEL': {
+      const levels = newState[action.stage - 1].levels;
+      const temp = levels.splice(action.levelA - 1, 1);
+      levels.splice(action.levelB - 1, 0, temp[0]);
+      updatePositions(levels);
+      break;
+    }
+    case 'ADD_GROUP': {
+      newState.push({
+        id: newStageId--,
+        flex_category: prompt('Enter new group name'),
+        name: prompt('Enter new stage name'),
```

Comment:
Having prompts in our reducer seems bad. We should prompt before getting here, and have the result of the prompts passed via our action.